src/parser/jobs/war/modules/SurgingTempest.tsx

```diff
@@ -1,25 +1,76 @@
 import {Trans} from '@lingui/react'
 import {DataLink} from 'components/ui/DbLink'
-import {Analyser} from 'parser/core/Analyser'
+import {ActionKey} from 'data/ACTIONS'
+import {Event} from 'event'
+import {filter} from 'parser/core/filter'
 import {dependency} from 'parser/core/Injectable'
 import {Actors} from 'parser/core/modules/Actors'
 import Checklist, {Rule, Requirement} from 'parser/core/modules/Checklist'
+import {Gauge as CoreGauge, TimerGauge} from 'parser/core/modules/Gauge'
 import {Invulnerability} from 'parser/core/modules/Invulnerability'
 import {Statuses} from 'parser/core/modules/Statuses'
+import Suggestions, {SEVERITY, TieredSuggestion} from 'parser/core/modules/Suggestions'
 import React from 'react'
 
-export class SurgingTempest extends Analyser {
+const SURGING_TEMPEST_GENERATORS: ActionKey[] = [
+	'MYTHRIL_TEMPEST',
+	'STORMS_EYE',
+	'INNER_RELEASE',
+]
+const SURGING_TEMPEST_GENERATION_AMOUNT = 30
+// Surging Tempest Extension
+const SURGING_TEMPEST_EXTENSION_AMOUNT = 10
+
+const ONE_MINUTE = 60000
+
+const SURGING_TEMPEST_EARLY_REFRESH_GRACE = 7500
+const STORMS_EYE_LOST_GAUGE = 10
+//TODO: Discuss with acc to select correct tier breakpoints
+const SUGGESTION_TIERS = {
+	1: SEVERITY.MINOR,
+	2: SEVERITY.MEDIUM,
+	5: SEVERITY.MAJOR,
+}
+
+export class SurgingTempest extends CoreGauge {
 	static override handle = 'surgingtempest'
 
 	@dependency private actors!: Actors
 	@dependency private checklist!: Checklist
 	@dependency private invulnerability!: Invulnerability
 	@dependency private statuses!: Statuses
+	@dependency private suggestions!: Suggestions
 
 	override initialise(): void {
+		this.addEventHook(
+			filter<Event>()
+				.source(this.parser.actor.id)
+				.action(this.data.matchActionId(SURGING_TEMPEST_GENERATORS)),
+			this.extendSurgingTempest
+		)
 		this.addEventHook('complete', this.onComplete)
 	}
 
+	private surgingTempest = this.add(new TimerGauge({
+		maximum: ONE_MINUTE,
+	}))
+
+	private earlyRefreshCount = 0
```

Comment:
Prefer to define vars before the init function. Also, var names:
```suggestion
	private surgeTimer = this.add(new TimerGauge({
		maximum: SURGE_MAX,
	}))

	private earlyEyes = 0
```